src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplGetCaseByIdTest.java

```diff
@@ -119,25 +125,94 @@ public void testGetCaseByCaseId_householdIndividualCase() throws CTPException {
     }
   }
 
+  @Test
+  public void testGetLatestFromCacheWhenResultsFromBothRmAndCache() throws Exception {
+    CaseContainerDTO caseFromCaseService = casesFromCaseService().get(0);
+    List<CachedCase> casesFromRepository = FixtureHelper.loadPackageFixtures(CachedCase[].class);
+    CachedCase cachedCase = casesFromRepository.get(0);
+
+    setUpIdsAndUprns(caseFromCaseService, cachedCase);
+
+    // Make sure that expected case has the most recent date
+    cachedCase.setCreatedDateTime(new Date());
+
+    doGetCaseById(caseFromCaseService, casesFromRepository, UUID_0, CACHED_CASE_UPRN_0);
+  }
+
+  @Test
+  public void testGetLatestFromCacheWhenResultsFromBothRmAndCacheWithSmallTimeDifference()
+      throws Exception {
+    CaseContainerDTO caseFromCaseService = casesFromCaseService().get(0);
+    List<CachedCase> casesFromRepository = FixtureHelper.loadPackageFixtures(CachedCase[].class);
+    CachedCase cachedCase = casesFromRepository.get(0);
+
+    setUpIdsAndUprns(caseFromCaseService, cachedCase);
+
+    // Make sure that expected case has the most recent date and that it is only 1 second closer
+    // than the next most recent date
+    cachedCase.setCreatedDateTime(utcDate(LocalDateTime.of(2020, 2, 3, 10, 4, 6)));
+    caseFromCaseService.setLastUpdated(utcDate(LocalDateTime.of(2020, 2, 3, 10, 4, 5)));
+
+    doGetCaseById(caseFromCaseService, casesFromRepository, UUID_0, CACHED_CASE_UPRN_0);
+  }
+
+  @Test
+  public void testGetLatestFromRmWhenResultsFromBothRmAndCacheWithSmallTimeDifferences()
+      throws Exception {
+    CaseContainerDTO caseFromCaseService = casesFromCaseService().get(0);
+    List<CachedCase> casesFromRepository = FixtureHelper.loadPackageFixtures(CachedCase[].class);
+    CachedCase cachedCase = casesFromRepository.get(0);
+
+    setUpIdsAndUprns(caseFromCaseService, cachedCase);
+
+    // Make sure that expected case has the most recent date and that it is only 1 second closer
+    // than the next most recent date
+    caseFromCaseService.setLastUpdated(utcDate(LocalDateTime.of(2020, 2, 3, 10, 4, 6)));
+    cachedCase.setCreatedDateTime(utcDate(LocalDateTime.of(2020, 2, 3, 10, 4, 5)));
```

Comment:
I found it hard to read the timestamp setup to see which case is the newest. It would be clearer if you had a baseTimestamp type variable of type LocalDateTime. One of the cases could use that (for passing in to utcDate()) and the other ought to be able to use it with a plusSeconds(1) or minusSeconds(1) adjustment.

Reply:
As this is a minor point I would rather leave it as is due to time pressures.